stream.go

```diff
@@ -243,11 +243,11 @@ func (s *stream) Write(p []byte) (int, error) {
 		s.mutex.Lock()
 	}
 
-	if err != nil {
```

Comment:
Everything below here can be replaced with `return s.err, len(p) - len(s.dataForWriting)`.